client/X11/xf_window.c

```diff
@@ -942,7 +944,11 @@ void xf_SetWindowVisibilityRects(xfContext* xfc, xfAppWindow* appWindow, RECTANG
 		xrects[i].height = rects[i].bottom - rects[i].top;
 	}
 
-	XShapeCombineRectangles(xfc->display, appWindow->handle, ShapeBounding, 0, 0, xrects, nrects, ShapeSet, 0);
+	/* Translate the server rail window offset to a local offset */
+	translatedWindowOffsetX = (appWindow->visibleOffsetX - appWindow->windowOffsetX);
+	translatedWindowOffsetY = (appWindow->visibleOffsetY - appWindow->windowOffsetY);
```

Comment:
Does this logic work correctly when "appWindow->windowOffsetX" or "appWindow->windowOffsetY" are negative(such that the appWindow->localWindowOffsetCorr" variables are non-zero)? I would just think those might need to be taken into account as those are used to translate those negative values into our local coordinate system.